jormungandr/src/explorer/graphql/mod.rs

```diff
@@ -1542,47 +1593,72 @@ impl Query {
             .collect()
     }
 
-    /// get the ref that the ledger currently considers as the main branch's
-    async fn tip(&self, context: &Context) -> Branch {
-        let (hash, state_ref) = context.db.get_tip().await;
+    /// get the block that the ledger currently considers as the main branch's
+    /// tip
+    async fn tip(&self, context: &Context<'_>) -> Branch {
+        let (hash, state_ref) = extract_context(&context).await.db.get_tip().await;
         Branch::from_id_and_state(hash, state_ref)
     }
 
-    async fn branch(&self, id: String, context: &Context) -> FieldResult<Branch> {
+    pub async fn branch(&self, context: &Context<'_>, id: String) -> FieldResult<Branch> {
```

Comment:
Looks a bit inconsistent that the branch ID is a `String`, but this PR does not change it.